turf/src/commonMain/kotlin/io/github/elcolto/geokjson/turf/booleans/Within.kt

```diff
@@ -0,0 +1,226 @@
+ */
+@Suppress("CyclomaticComplexMethod")
+@ExperimentalTurfApi
+@Throws(IllegalStateException::class)
+public fun within(geometry1: Geometry, geometry2: Geometry): Boolean = when (geometry1) {
+    is Point -> when (geometry2) {
+        is MultiPoint -> isPointInMultiPoint(geometry1, geometry2)
+        is LineString -> pointOnLine(geometry1, geometry2, ignoreEndVertices = true)
+        is Polygon -> pointInPolygon(geometry1, geometry2, ignoreBoundary = true)
+        is MultiPolygon -> pointInPolygon(geometry1, geometry2, ignoreBoundary = true)
+        is GeometryCollection,
+        is MultiLineString,
+        is Point,
+        -> error("geometry2 ${geometry2::class.simpleName} is not supported")
+    }
+
+    is MultiPoint -> when (geometry2) {
+        is MultiPoint -> isMultiPointInMultiPoint(geometry1, geometry2)
+        is LineString -> isMultiPointOnLine(geometry1, geometry2)
+        is Polygon -> isMultiPointInPoly(geometry1, geometry2)
+        is MultiPolygon -> isMultiPointInPoly(geometry1, geometry2)
+        is MultiLineString,
+        is GeometryCollection,
+        is Point,
+        -> error("geometry2 ${geometry2::class.simpleName} is not supported")
+    }
+
+    is LineString -> when (geometry2) {
+        is LineString -> isLineOnLine(geometry1, geometry2)
+        is Polygon -> isLineInPoly(geometry1, geometry2)
+        is MultiPolygon -> isLineInPoly(geometry1, geometry2)
+        is GeometryCollection,
+        is MultiLineString,
+        is MultiPoint,
+        is Point,
+        -> error("geometry2 ${geometry2::class.simpleName} is not supported")
+    }
+
+    is Polygon -> when (geometry2) {
+        is Polygon -> isPolyInPoly(geometry1, geometry2)
+        is MultiPolygon -> isPolyInPoly(geometry1, geometry2)
+        is GeometryCollection,
+        is LineString,
+        is MultiLineString,
+        is MultiPoint,
+        is Point,
+        -> error("geometry2 ${geometry2::class.simpleName} is not supported")
+    }
+
+    is GeometryCollection,
+    is MultiLineString,
+    is MultiPolygon,
+    -> error("geometry1 ${geometry1::class.simpleName} is not supported")
+}
+
+private fun isPointInMultiPoint(point: Point, multiPoint: MultiPoint) =
+    multiPoint.coordinates.any { position -> position == point.coordinates }
+
+private fun isMultiPointInMultiPoint(multiPoint1: MultiPoint, multiPoint2: MultiPoint) =
+    multiPoint2.coordinates.containsAll(multiPoint1.coordinates)
+
+@OptIn(ExperimentalTurfApi::class)
+private fun isMultiPointOnLine(multiPoint: MultiPoint, lineString: LineString): Boolean {
+    var foundInsidePoint = false
+
+    multiPoint.coordinates.forEach { position ->
+        if (!pointOnLine(position, lineString)) {
+            return false
+        }
+
+        if (!foundInsidePoint) {
+            foundInsidePoint = pointOnLine(position, lineString, ignoreEndVertices = true)
+        }
+    }
+
+    return foundInsidePoint
```

Comment:
Thought so as well in first place. And i guess this would be the easier implementation, however this approach would require 2 iterations over the coordinates whereas the current one only requires one single iteration.

Reply:
I changed it to the simpler approach